app/models/user.rb

```diff
@@ -240,6 +244,24 @@ def activate!
     update_columns(suspended_at: nil)
   end
 
+  def self.google_login_domain?(email)
+    email = email.downcase
+    email_host = email.include?('@') ? email.split("@")[-1] : email
+    Devise.omniauth_configs[:google_oauth2].options[:hd].include?(email_host)
+  end
+
+  def self.from_omniauth(auth_hash)
+    return nil unless Rails.configuration.google_login_enabled
+    return nil unless auth_hash['provider'] == "google_oauth2"
+
+    email = auth_hash.info['email']
+    return nil unless google_login_domain?(email) && google_login_domain?(auth_hash.extra.id_info["hd"])
```

Comment:
what extra do we get from verifying this `hd` in the auth_hash ? -- seems to me since Google gave us a callback, they believe `whoever@google-oauth-supported-domain.org` is verified to be a google user so we should be able to just validate the hostname of that if we want

but I could be wrong!!

Reply:
The `hd` check verifies that they're coming in from one of the Google hosted domains (i.e., gsuite domains) in question.

I believe we get nothing or very close to nothing from doing the additional verification here. However, I'm always nervous about login integrations. I figure in the unlikely case we misunderstood the meaning of `email` and how much verification is applied to it, we can at least be sure the only people who can access this codepath are logged into an @codeforamerica or @getyourrefund Google account (i.e. in the relevant hosted domain).